classes/OJSSwordDeposit.inc.php

```diff
@@ -0,0 +1,202 @@
+// 			}
+// 		}
+	}
+
+	/**
+	 * Add a file to a package. Used internally.
+	 * @param $submissionFile SubmissionFile
+	 */
+	public function _addFile($submissionFile) {
+		$targetFilename = $this->_outPath . '/files/' . $submissionFile->getLocalizedName();
+		copy($submissionFile->getFilePath(), $targetFilename);
+		$this->_package->addFile($submissionFile->getLocalizedName(), $submissionFile->getFileType());
+	}
+
+	/**
+	 * Add all article galleys to the deposit package.
+	 */
+	public function addGalleys() {
+		foreach ($this->_article->getGalleys() as $galley) {
+			$this->_addFile($galley->getFile());
+		}
+	}
+
+	/**
+	 * Add the single most recent editorial file to the deposit package.
+	 * @return boolean true iff a file was successfully added to the package
+	 */
+	public function addEditorial() {
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
+		$submissionFiles = $submissionFileDao->getBySubmissionId($this->_article->getId());
+		// getBySubmission orders results by id ASC, let's reverse the array to have recent files first
+		$submissionFiles = array_reverse($submissionFiles, true);
+		$files = array();
+		foreach ($submissionFiles as $submissionFile) {
+			$fileStage = $submissionFile->getFileStage();
+			if (!isset($files[$fileStage])) {
+				$files[$fileStage] = array();
+			}
+			$files[$fileStage][] = $submissionFile;
+		}
+		// Move through stages in reverse order and try to use them.
+		$stages = array(
+			SUBMISSION_FILE_PRODUCTION_READY,
+			SUBMISSION_FILE_COPYEDIT,
+			SUBMISSION_FILE_REVIEW_FILE,
+			SUBMISSION_FILE_SUBMISSION
+		);
+		$mostRecentEditorialFile = null;
+		foreach ($stages as $subFileStage) {
+			if (isset($files[$subFileStage])) {
+				$mostRecentEditorialFile = array_shift($files[$subFileStage]);
+				$this->_addFile($mostRecentEditorialFile);
+				return true;
+			}
+		}
+		// TODO when the above didn't work, SWORD plugin for OJS2 was trying "the editor version"
+		// how does it apply to OJS3? (I found no SectionEditorSubmissionDAO in the code base)
+		return false;
+	}
+
+	/**
+	 * Build the package.
+	 */
+	public function createPackage() {
+		return $this->_package->create();
+	}
+
+	/**
+	 * Deposit the package.
+	 * @param $url string SWORD deposit URL
+	 * @param $username string SWORD deposit username (i.e. email address for DSPACE)
+	 * @param $password string SWORD deposit password
+	 */
+	public function deposit($url, $username, $password) {
+		$client = new SWORDAPPClient();
+		$response = $client->deposit(
+			$url, $username, $password,
+			'',
+			$this->_outPath . '/deposit.zip',
+			'http://purl.org/net/sword/package/METSDSpaceSIP',
+			'application/zip', false, true
+		);
+		return $response;
+	}
+
+	/**
+	 * Clean up after a deposit, i.e. removing all created files.
+	 */
+	public function cleanup() {
+		import('lib.pkp.classes.file.FileManager');
+		$fileManager = new FileManager();
+		$fileManager->rmtree($this->_outPath);
+	}
+}
+
+?>
```

Comment:
I see you've removed the closing `?>` elsewhere; best to do it throughout.

Reply:
removed.